### PR TITLE
YETUS-858. Add support for Golang

### PR DESCRIPTION
**Pull request by aw-was-here**

### CODE REVIEW

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/1/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 230 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 87 | master passed |
| +1 | javadoc | 21 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 7 | Maven dependency ordering for patch |
| +1 | mvninstall | 21 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 1 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 3 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 20 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 10 | precommit in the patch passed. |
| +1 | unit | 9 | asf-site-src in the patch passed. |
| +1 | asflicense | 18 | The patch does not generate ASF License warnings. |
| | | 477 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/50 |
| Optional Tests | dupname asflicense markdownlint shellcheck shelldocs javac javadoc unit hadolint |
| uname | Linux 8880d0e9be7d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 8df783b |
| Default Java | 1.8.0_191 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/1/testReport/ |
| Max. process+thread count | 100 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT http://yetus.apache.org |


This message was automatically generated.

File: precommit/src/main/shell/test-patch.d/checkmake.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SHELLDOC-IGNORE
+
+add_test_type checkmake
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/checkmake.sh]: how is this related to go?

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/2/console in case of problems.

File: precommit/src/main/shell/test-patch.d/checkmake.sh (hunk 1 above)
Comment (aw-was-here) [reply precommit/src/main/shell/test-patch.d/checkmake.sh]: Pretty much every project that is in Go uses make as a driver.  Plus checkmake itself is written in Go.

**Comment by aw-was-here** [comment]
Updated to use Go 1.12

**Comment by aw-was-here** [comment]
> you already have an example go project in mind for testing this against?

Publically, I've been using LF Edge EVE:

* https://github.com/zededa/eve

My Circle CI runs are here:

* https://circleci.com/gh/aw-was-here/eve

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 352 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 88 | master passed |
| +1 | javadoc | 22 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 6 | Maven dependency ordering for patch |
| +1 | mvninstall | 19 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 1 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 2 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 16 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 9 | precommit in the patch passed. |
| +1 | unit | 7 | asf-site-src in the patch passed. |
| +1 | asflicense | 16 | The patch does not generate ASF License warnings. |
| | | 589 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/50 |
| Optional Tests | dupname asflicense markdownlint shellcheck shelldocs javac javadoc unit hadolint |
| uname | Linux f3556c5329ec 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 5aca27b |
| Default Java | 1.8.0_191 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/2/testReport/ |
| Max. process+thread count | 99 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/2/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT http://yetus.apache.org |


This message was automatically generated.

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/3/console in case of problems.

**Comment by aw-was-here** [comment]
Rebase
Add go.mod to whitespace exemption

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 330 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 88 | master passed |
| +1 | javadoc | 18 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 5 | Maven dependency ordering for patch |
| +1 | mvninstall | 18 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 1 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 2 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 15 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8 | precommit in the patch passed. |
| +1 | unit | 8 | asf-site-src in the patch passed. |
| +1 | asflicense | 15 | The patch does not generate ASF License warnings. |
| | | 556 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/50 |
| Optional Tests | dupname asflicense markdownlint shellcheck shelldocs javac javadoc unit hadolint |
| uname | Linux 331f9f6d2c70 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 5aca27b |
| Default Java | 1.8.0_191 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/3/testReport/ |
| Max. process+thread count | 100 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT http://yetus.apache.org |


This message was automatically generated.

File: precommit/src/main/shell/test-patch.d/revive.sh [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  pushd "${BASEDIR}" >/dev/null || return 1
+
+  args=('-formatter' 'default')
+  if [[ -f "${REVIVE_CONFIG}" ]]; then
+    args+=('-config' "${REVIVE_CONFIG}")
+  fi
+
+  for i in "${CHANGED_FILES[@]}"; do
+    if [[ ${i} =~ \.go$ && -f ${i} ]]; then
+      "${REVIVE}" "${args[@]}" "${i}" | sort -t : -k 2 -k 3 -n >> "${PATCH_DIR}/${repostatus}-revive-result.txt"
```
Comment (aw-was-here) [inline precommit/src/main/shell/test-patch.d/revive.sh]: This sort is wrong. :(

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/4/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 411 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 109 | master passed |
| +1 | javadoc | 22 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 6 | Maven dependency ordering for patch |
| +1 | mvninstall | 23 | the patch passed |
| +1 | hadolint | 4 | There were no new hadolint issues. |
| +1 | markdownlint | 1 | There were no new markdownlint issues. |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | shelldocs | 2 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 19 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 11 | precommit in the patch passed. |
| +1 | unit | 9 | asf-site-src in the patch passed. |
| +1 | asflicense | 20 | The patch does not generate ASF License warnings. |
| | | 695 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/50 |
| Optional Tests | dupname asflicense markdownlint shellcheck shelldocs javac javadoc unit hadolint |
| uname | Linux 258a60501fde 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / ebdf1ea |
| Default Java | 1.8.0_191 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/4/testReport/ |
| Max. process+thread count | 103 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-50/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT http://yetus.apache.org |


This message was automatically generated.

**Comment by aw-was-here** [comment]
I'm going to close this for now.  There are some other things that need to get added first.
